src/features/search/components/SearchResults/SearchResults.tsx

```diff
@@ -110,6 +114,16 @@ export const SearchResults: React.FC = () => {
     [nbHits, searchState]
   )
 
+  useEffect(() => {
+    const shouldRefetchResults = Boolean(
+      (userPosition && !previousUserPosition) || (!userPosition && previousUserPosition)
+    )
+
+    if (shouldRefetchResults) {
+      refetch()
+    }
+  }, [previousUserPosition, refetch, userPosition])
```

Comment:
Quitte à le mettre dans une variable, tu aurais pu alors sortir la variable pour te débarasser d'une dépendance, ç'aurait peut-être été plus clair. Mais pas de souci.

```suggestion
  const shouldRefetchResults = Boolean(
    (userPosition && !previousUserPosition) || (!userPosition && previousUserPosition)
  ) // ou useMemo mais pas nécessaire

  useEffect(() => {
    if (shouldRefetchResults) {
      refetch()
    }
  }, [refetch, shouldRefetchResults])
```